engine/src/control.cpp

```diff
@@ -1113,6 +1113,7 @@ void MCControl::drawselected(MCDC *dc)
 	else
 		dc->setfillstyle(FillSolid, nil, 0, 0);
 	dc->setforeground(MCselectioncolor);
+    dc->setquality(QUALITY_SMOOTH);
```

Comment:
You might need to stack the current version of quality and then reset before exiting the paint handler - otherwise it will affect any subsequent drawing to the context. (I believe the graphic object does this when handling the 'antialiased' property).